internal/receiver/smartagentreceiver/converter/traces.go

```diff
@@ -24,12 +25,44 @@ import (
 	"go.uber.org/zap"
 )
 
+const (
+	sfxDataSourceKey         = "data_source_ip"
+	resourceClientIPAttrName = "ip"
+)
+
 var zipkinv2Translator = zipkinv2.ToTranslator{ParseStringTags: false}
 
 func sfxSpansToPDataTraces(spans []*trace.Span, logger *zap.Logger) (pdata.Traces, error) {
-	// SFx trace is effectively zipkin, so more convenient to convert to it and then rely
-	// on existing zipkin receiver translator
-	return zipkinv2Translator.ToTraces(sfxToZipkinSpans(spans, logger))
+	// we batch sfx spans by the client IP that reported the spans to collector. Each
+	// batch gets translated separately to ensure that spans from sources with different
+	// IPs don't get bundled together under the same resource.
+	batches := map[string][]*trace.Span{}
+	for _, span := range spans {
+		var sourceIP string
+		if val, ok := span.Meta[sfxDataSourceKey]; ok {
+			if ip, ok := val.(net.IP); ok {
+				sourceIP = ip.String()
+			}
+		}
+		batches[sourceIP] = append(batches[sourceIP], span)
+	}
+
+	traces := pdata.NewTraces()
+	rss := traces.ResourceSpans()
+	for ip, s := range batches {
+		// SFx trace is effectively zipkin, so more convenient to convert to it and then rely
+		// on existing zipkin receiver translator
+		translated, err := zipkinv2Translator.ToTraces(sfxToZipkinSpans(s, logger))
+		if err != nil {
+			return traces, err
```

Comment:
Since this returns `traces` with partial results in case of error should it combine all errors and return all spans that it could process? Alternatively it could do the traditional: `return nil, err`.